core/include/detray/definitions/detail/accessor.hpp

```diff
@@ -73,5 +83,21 @@ DETRAY_HOST_DEVICE constexpr tuple_type<unwrap_decay_t<Types>...> make_tuple(
     return tuple_type<unwrap_decay_t<Types>...>(std::forward<Types>(args)...);
 }
 
+template <typename mask_store_t,
+          std::enable_if_t<std::is_class_v<typename std::remove_reference_t<
```

Comment:
I think this should be defined somwhere else

Reply:
Hm, I moved it to the intersection kernel now

Reply:
I think moving it into mask_store is more reasonable?

Reply:
But one of them should explicitly be used when we don't use a mask store, so I have to leave that one in another file

Reply:
makes sense